csharp/Microsoft.Azure.Devices.Client.NetMF/Common/Security/SharedAccessSignatureBuilder.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    using Microsoft.Azure.Devices.Client.Extensions;
+    using System;
+    using System.Text;
+
+    sealed class SharedAccessSignatureBuilder
+    {
+        string key;
+
+        public SharedAccessSignatureBuilder()
+        {
+            this.TimeToLive = new TimeSpan(0, 20, 0);
+        }
+
+        public string KeyName { get; set; }
+
+        public string Key
+        {
+            get
+            {
+                return this.key;
+            }
+
+            set
+            {
```

Comment:
EnsureBase64String call was removed. Can we leave the call in and "#if" the implementation of the method. That way we can go back and figure out implementing this without the use of Generics.